spec/src/main/asciidoc/entities.asciidoc

```diff
@@ -18,42 +18,80 @@
 
 == GraphQL Entities
 
+Entities are the objects used in GraphQL. The can be simple objects ("scalars" in GraphQL terminology) or more complex objects
+that are composed of scalars.
+According to the https://graphql.org/learn/schema/#scalar-types[GraphQL documentation] a scalar has no sub-fields, and all
+GraphQL implementations are expected to handle, the following scalar types:
+- `Int` - which maps to a Java `int`/`Integer` or `long`/`Long`.
+- `Float` - which maps to a Java `float`/`Float` or `double`/`Double`.
+- `String` - which maps to a Java `char`, `char[]`, `String`, etc.
+- `Boolean` - which maps to a Java `boolean`/`Boolean`.
+- `ID` - which is a specialized type serialized like a `String`.
+
+In order for an entity class to be defined in the GraphQL schema, it must meet at least one of the following criteria:
+- It must be the return type or parameter of a query or mutation method,
+- It must be annotated with `@Type`,
+- It must be annotated with `@InputType`
+
 === POJOs
 
-TODO: info about basic entities based on JSON-B
+Any Plain Old Java Object (POJO) can be an entity.  No special annotations are required. Implementations of MicroProfile
+GraphQL must use JSON-B to serialize and deserialize entities to JSON, so it is possible to further define entities using
+JSON-B annotations.
+
+If the entity cannot be serialized by JSON-B, the implementation must return in an internal server error to the client.
 
 === Types vs InputTypes
 
-TODO: info on types and input types
+GraphQL differentiates types from input types.  Input types are entities that are sent by the client as arguments to queries or
+mutations. Types are entities that are sent from the server to the client as return types from queries or mutations.
+
+In many cases the same Java type can be used for input (sent _from_ client) and output (sent _to_ the client), but there are
+cases where an application may need two different Java types to handle input and output.
+
+The `@Type` annotation is used for output entities while the `@InputType` annotation is used for input entities.
```

Comment:
Now that I read it like this, would it not be more consistent to call @Type the @OutputType or @ReturnType ??

Reply:
We could.  I think the names `@Type` and `@InputType` were chosen due to the language used in GraphQL.  In the schema, entities that are returned are labeled like `type MyType` whereas entities that are passed to queries/mutations are labeled like `input MyInputType`.

Would it help if I re-worded this somewhat?  Or maybe provided a link to the GraphQL docs (like [this page](https://graphql.org/learn/schema/) ) that explains the types wrt to the schema?

Reply:
OK cool, makes sense. So maybe one should be @Type and the other @Input ??
For now we can merge and maybe discuss in Gitter or add it to the agenda for next week ?